src/app/selectors.js

```diff
@@ -154,6 +153,29 @@ const groupModelsByStatus = modelData => {
   return grouped;
 };
 
+/**
+  Returns a grouped collection of model statuses by owner.
+  @param {Object} modelData
+  @returns {Function} The grouped model statuses by owner.
+*/
+const groupModelsByOwner = modelData => {
+  const grouped = {};
+  if (!modelData) {
+    return grouped;
+  }
+  for (let modelUUID in modelData) {
+    const model = modelData[modelUUID];
+    if (model.info) {
+      const owner = extractOwnerName(model.info.ownerTag);
```

Comment:
Correct, but the `ownerTag` is a required field in the API response.

Reply:
Why check for model.info then?

Reply:
Ahh, because with the current way we have to make multiple calls to populate the full model details the `.info` path may or may not be available when this selector is run, but if it does exist then it'll be fully populated.

You can see that here: https://github.com/canonical-web-and-design/jaas-dashboard/blob/master/src/juju/reducers.js#L52

I can of course explain the architecture in more detail offline if you'd like.